src/components/TicTacToe.jsx

```diff
@@ -187,123 +157,76 @@ class TicTacToe extends Component {
-      boxFiveValue,
-      boxSixValue,
-      boxSevenValue,
-      boxEightValue,
-      boxNineValue,
-      win } = this.state;
+    const { box, winner } = this.state;
     return (
       <div className="container">
         <header className="title" style={style.titleStyle}>TIC TAC TOE</header>
         {
-          win && <h2 style={{ textAlign: 'center' }}>You have won the game!</h2>
+          winner && <h2 style={{ textAlign: 'center' }}>{winner} has won the game!</h2>
         }
-        <ul className="dashboard" style={style.dashboardStyle}>
-          <li
+        <ul className="dashboard" style={style.dashboardStyle}> 
+          <Box
+            key={0}
             style={style.cardBoardEven}
-            ref={input => this.boxOneInput = input}
-            onClick={() => this.handleClick('One')}>{boxOneValue}</li>
-          <li
+            onClick={() => this.handleClick(0)}
+          >{box[0]}</Box>
+          <Box
+            key={1}
             style={style.cardBoardOdd}
-            ref={input => this.boxTwoInput = input}
-            onClick={() => this.handleClick('Two')}>{boxTwoValue}</li>
-          <li
+            onClick={() => this.handleClick(1)}
+          >{box[1]}</Box>
+          <Box
+            key={2}
             style={style.cardBoardEven}
-            ref={input => this.boxThreeInput = input}
-            onClick={() => this.handleClick('Three')}>{boxThreeValue}</li>
-          <li
+            onClick={() => this.handleClick(2)}
+          >{box[2]}</Box>
+          <Box
+            key={3}
             style={style.cardBoardOdd}
-            ref={input => this.boxFourInput = input}
-            onClick={() => this.handleClick('Four')}>{boxFourValue}</li>
-          <li
+            onClick={() => this.handleClick(3)}
+          >{box[3]}</Box>
+          <Box
+            key={4}
             style={style.cardBoardEven}
-            ref={input => this.boxFiveInput = input}
-            onClick={() => this.handleClick('Five')}>{boxFiveValue}</li>
-          <li
+            onClick={() => this.handleClick(4)}
+          >{box[4]}</Box>
+          <Box
+            key={5}
             style={style.cardBoardOdd}
-            ref={input => this.boxSixInput = input}
-            onClick={() => this.handleClick('Six')}>{boxSixValue}</li>
-          <li
+            onClick={() => this.handleClick(5)}
+          >{box[5]}</Box>
+          <Box
+            key={6}
             style={style.cardBoardEven}
-            ref={input => this.boxSevenInput = input}
-            onClick={() => this.handleClick('Seven')}>{boxSevenValue}</li>
-          <li
+            onClick={() => this.handleClick(6)}
+          >{box[6]}</Box>
+          <Box
+            key={7}
             style={style.cardBoardOdd}
-            ref={input => this.boxEightInput = input}
-            onClick={() => this.handleClick('Eight')}>{boxEightValue}</li>
-          <li
+            onClick={() => this.handleClick(7)}
```

Comment:
Hmmn you binded in your constructor. yet you are binding i the view too. This would lead to a lot of performance issues moving forward moving forward you could pass
{this.handleClick} and use it with your mapValues